eng/targets/Java.Common.targets

```diff
@@ -0,0 +1,87 @@
+<Project DefaultTargets="Build">
+
+  <PropertyGroup>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' == ''">$(MSBuildProjectDirectory)\obj\</BaseIntermediateOutputPath>
+    <IntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)'))$(Configuration)\</IntermediateOutputPath>
+    <BuildDependsOn>
+      PrepareForBuild;
+      ResolveProjectReferences;
+      _Build;
+    </BuildDependsOn>
+    <JavaBuildArgs Condition="'$(JavaBuildArgs)' == ''">../gradlew $(GradleOptions) compileJava</JavaBuildArgs>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <JavaFiles Include="src\**\*.java" />
+    <JavaFiles Include="*.javaproj" />
+    <JavaFiles Include="build.gradle" />
+
+    <BuildOutputFiles Include="$(BaseIntermediateOutputPath)build-sentinel" />
+  </ItemGroup>
+
+  <Target Name="Restore" />
+
+  <PropertyGroup>
+    <PackDependsOn Condition=" '$(NoBuild)' != 'true' ">
+      $(PackDependsOn);
+      Build
+    </PackDependsOn>
+  </PropertyGroup>
+
+  <Target Name="PrepareForBuild">
+    <MakeDir Directories="$(IntermediateOutputPath);$(PackageOutputPath)"  />
+  </Target>
+
+  <Target Name="ResolveProjectReferences">
+    <MSBuild Projects="@(ProjectReference)"
+             BuildInParallel="true" />
+  </Target>
+
+  <Target Name="Build" DependsOnTargets="$(BuildDependsOn)" />
+
+  <Target Name="GetBuildInputCacheFile">
+    <Hash ItemsToHash="@(JavaFiles)">
+      <Output TaskParameter="HashResult" PropertyName="_JavaFileHash" />
+    </Hash>
+
+    <WriteLinesToFile
+      Lines="$(_JavaFileHash)"
+      File="$(BaseIntermediateOutputPath)javafiles.cache"
+      Overwrite="True"
+      WriteOnlyWhenDifferent="True" />
+  </Target>
+
+  <Target Name="_Build"
+    Condition="'$(IsBuildable)' != 'false'"
+    DependsOnTargets="GetBuildInputCacheFile"
+    Inputs="@(JavaFiles);$(BaseIntermediateOutputPath)javafiles.cache"
+    Outputs="@(BuildOutputFiles)">
+    <Telemetry EventName="NETCORE_ENGINEERING_TELEMETRY" EventData="Category=Build" />
+    <Exec Command="$(JavaBuildArgs)" />
+    <WriteLinesToFile Overwrite="true" File="$(BaseIntermediateOutputPath)build-sentinel" />
+  </Target>
+
+  <PropertyGroup>
+    <PackDependsOn>
+      $(PackDependsOn);
+      PrepareForBuild
+    </PackDependsOn>
+    <PackDependsOn Condition="'$(NoBuild)' != 'true'">
+      $(PackDependsOn);
+      Build;
+    </PackDependsOn>
```

Comment:
With the current settings, `$(PackDependsOn)` will contain at least `Build;PrepareForBuild;Build`. Suspect that means the earlier setting is the one that should be removed.